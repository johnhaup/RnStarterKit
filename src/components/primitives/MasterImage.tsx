import React from 'react';
import { Image, ImageStyle, Platform } from 'react-native';
import FastImage from 'react-native-fast-image';

interface Props {
  uri?: string | undefined | null;
  disableFast?: boolean;
  contain?: boolean;
  style?: ImageStyle | ImageStyle[];
}

export const MasterImage = ({
  uri,
  disableFast,
  contain,
  style = {},
}: Props) => {
  if (!uri) {
    return null;
  }

  const resizeMode = contain ? 'contain' : 'cover';
  // Android throws error with FastImage
  const useStandardImage = disableFast || Platform.OS === 'android';

  return useStandardImage ? (
    <Image style={style} source={{ uri }} resizeMode={resizeMode} />
  ) : (
    <FastImage style={style} source={{ uri }} resizeMode={resizeMode} />
  );
};
