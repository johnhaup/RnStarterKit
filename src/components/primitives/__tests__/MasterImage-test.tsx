/**
 * @format
 */

import React from 'react';
import 'react-native';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import { MasterImage } from '../MasterImage';

it('renders correctly', () => {
  const snap = renderer.create(
    <MasterImage
      uri={
        'https://www.udiscovermusic.com/wp-content/uploads/2019/04/Nirvana-Live-At-Reading-press-shot-02-CREDIT-Charles-Peterson-web-optimised-1000.jpg'
      }
    />,
  );
  expect(snap).toMatchSnapshot();
});

it('returns null when no uri provided', () => {
  const imageWithoutUri = renderer.create(<MasterImage />).getInstance();
  expect(imageWithoutUri).toEqual(null);
});
