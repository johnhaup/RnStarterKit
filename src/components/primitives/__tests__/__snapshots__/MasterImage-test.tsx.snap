// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "overflow": "hidden",
      },
      Object {},
    ]
  }
>
  <FastImageView
    resizeMode="cover"
    source={
      Object {
        "uri": "https://www.udiscovermusic.com/wp-content/uploads/2019/04/Nirvana-Live-At-Reading-press-shot-02-CREDIT-Charles-Peterson-web-optimised-1000.jpg",
      }
    }
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  />
</View>
`;
